### PR TITLE
Update histogram-aggregation docs

### PR DESCRIPTION
**Pull request by philippkahr**
Without specifiying the mapping explicitly it will just be stored as longs and then the histogram function doesn't really show it's cool purpose. As discovered by @herrBez 

If I just run the PUT command I get this:

```
{
  "metrics_index": {
    "aliases": {},
    "mappings": {
      "properties": {
        "latency_histo": {
          "properties": {
            "counts": {
              "type": "long"
            },
            "values": {
              "type": "long"
            }
          }
        },
        "network": {
          "properties": {
            "name": {
              "type": "text",
              "fields": {
                "keyword": {
                  "type": "keyword",
                  "ignore_above": 256
                }
              }
            }
          }
        }
      }
    }
}
```

Also the documentation mentions `latency_histo`. as a field however the example answer uses a bucket called `price`, which makes even less sense...

I modified it to work as expected

### CODE REVIEW

**Comment by philippkahr** [comment]
Will recreate on main

File: docs/reference/aggregations/bucket/histogram-aggregation.asciidoc [hunk 1]
```diff
@@ -332,6 +332,25 @@ with latency metrics (in milliseconds) for different networks:
...
+  "mappings": {
+    "properties": {
+      "network": {
+        "properties": {
+          "name": {
+            "type": "keyword"
+          }
+        }
+      },
+      "latency_hist": {
```
Comment (abdonpijpelink) [inline docs/reference/aggregations/bucket/histogram-aggregation.asciidoc:346]:
```suggestion
      "latency_histo": {
```

File: docs/reference/aggregations/bucket/histogram-aggregation.asciidoc [hunk 2]
```diff
@@ -332,6 +332,25 @@ with latency metrics (in milliseconds) for different networks:
...
+         "type": "histogram"
+        }
+      }
+    }
+  }
+}
```
Comment (abdonpijpelink) [inline docs/reference/aggregations/bucket/histogram-aggregation.asciidoc:352]:
```suggestion
         "type": "histogram"
      }
    }
  }
}
```

Comment (abdonpijpelink) [reply docs/reference/aggregations/bucket/histogram-aggregation.asciidoc:352]: One closing curly bracket too many

File: docs/reference/aggregations/bucket/histogram-aggregation.asciidoc [hunk 3]
```diff
@@ -371,29 +390,28 @@ return the following output:
...
+          "key": 15,
+          "doc_count": 1
         }
       ]
     }
-  }
-}
+...
 --------------------------------------------------
 // TESTRESPONSE[skip:test not setup]
```
Comment (abdonpijpelink) [inline docs/reference/aggregations/bucket/histogram-aggregation.asciidoc:416]: I think we should add a test here. This response is incorrect, and that would have been caught by having a test. Let me know if you need a hand with that.

**Review by abdonpijpelink — CHANGES_REQUESTED** [review]
Thanks for fixing this @philippkahr !

I've left some suggestions.

**Review by abdonpijpelink — APPROVED** [review]

**Comment by abdonpijpelink** [comment]
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.9|Cherrypick failed because the selected commit (4ccc5a9c) is empty. Did you already backport this commit? |
|✅|8.8|[](https://github.com/elastic/elasticsearch/pull/97151)|

### Manual backport
To create the backport manually run:
```
backport --pr 96974
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**